Add rebuilderd role

kpcyrd opened the pull request:
- [x] deploy a secret so `rebuilderd-sync` can authenticate with rebuilderd
- [x] deploy a worker
- [ ] configure rebuilderd in nginx proxy

kpcyrd committed:
Add basic rebuilderd role

SantiagoTorres commented on ansible/roles/purdue-rebuilderd/templates/rebuilderd.conf:
```diff
@@ -0,0 +1,40 @@
+## rebuildctl is searching for this cookie in ~/.config/rebuilderd.conf, /etc/rebuilderd.conf and
+## /var/lib/rebuilderd/auth-cookie in that order.
+#[auth]
+#cookie = "INSECURE"
```
I think other than this being templatable this PR looks good right?

kpcyrd replied:
That'd work. There's currently no vault yet, so if I'd go ahead and create one everybody would need to set it up as well or they won't be able to deploy anymore.

SantiagoTorres replied:
yeah, I think that's ok. I think we set a passphrase for the vault already right?

kpcyrd replied:
Yes, I'm already using it and documented how to configure it in the readme. :)

SantiagoTorres reviewed (commented):
overall, shall we move the cookie bit to a different PR and go ahead? what do you think @hojoung97 ?

hojoung97 commented on ansible/roles/purdue-rebuilderd/files/rebuilderd.service:
```diff
@@ -0,0 +1,17 @@
+[Unix]
```
I think this is a small typo

hojoungjang commented on ansible/roles/purdue-rebuilderd/files/rebuilderd.service:
```diff
@@ -0,0 +1,17 @@
+[Unix]
```
I think this is a small typo

kpcyrd replied:
Good catch, thanks!

hojoungjang reviewed (commented):
@SantiagoTorres @kpcyrd  Hello, thanks for looping me in. I had a chance to look. I think this looks great and honestly, I am mostly learning from you guys :)

kpcyrd committed:
Setup vault for rebuilderd secrets

kpcyrd commented:
I've added an encrypted `rebuilderd_auth_cookie` and `rebuilderd_signup_secret` secret to the repo, the vault key is expected to be located at `ansible/vault_key`. This is also documented in the README.

I've also fixed a few minor mistakes and tested the import works. There's now a backlog of packages waiting to be rebuilt:
```
core@localhost ~ $ docker exec rebuilderd rebuildctl queue ls
2021-11-06 18:26:22 libglib2.0-bin 2.70.1-1                                                                      "debian" "main" "amd64"
2021-11-06 18:26:22 libglib2.0-dev 2.70.1-1                                                                      "debian" "main" "amd64"
2021-11-06 18:26:22 dfrs 0.0.6-3                                                                                 "debian" "main" "amd64"
2021-11-06 18:26:22 sniffglue 0.14.0-1                                                                           "debian" "main" "amd64"
2021-11-06 18:26:22 anarchism 15.3-3                                                                             "debian" "main" "all"
2021-11-06 18:26:22 binutils-arm-none-eabi 15                                                                    "debian" "main" "amd64"
2021-11-06 18:26:22 librust-sniffglue-dev 0.14.0-1                                                               "debian" "main" "amd64"
core@localhost ~ $ 
```

kpcyrd committed:
Deploy rebuilderd-worker for debian

kpcyrd commented:
The change could be merged as-is imo, I can look into the status page in a follow-up PR.

SantiagoTorres commented:
LGTM!
